pyiron_contrib/tinybase/murn.py

```diff
@@ -0,0 +1,94 @@
+from pyiron_contrib.tinybase.container import (
+            AbstractOutput,
+            StructureInput,
+)
+from pyiron_contrib.tinybase.node import (
+            AbstractNode,
+            ListNode,
+            ListInput,
+            ReturnStatus
+)
+
+from copy import deepcopy
+
+import numpy as np
+import matplotlib.pyplot as plt
+import scipy.interpolate as si
+import scipy.optimize as so
+
+from pyiron_atomistics.atomistics.structure.has_structure import HasStructure
+
+MurnaghanInputBase = StructureInput.from_attributes(
+        "MurnaghanInputBase",
+        "strains",
+        "node"
+)
+
+class MurnaghanInput(MurnaghanInputBase, ListInput):
+    def check_ready(self):
+        structure_ready = self.structure is not None
+        strain_ready = len(self.strains) > 0
+        node = self.node
+        node.input.structure = self.structure
```

Comment:
This seems like an action, not a check...

Reply:
This is just done so that `node.check_ready` can be called to so if the "sub-node" is ready. The whole input checking logic is quite weak though, so maybe there's a better way for this.

Reply:
I'm afraid I'm still not sure I follow, but I'll keep reading the code.

The fact you invoked "sub-node readiness" does, however, add to my complaint about code-smelliness with having nodes take nodes as input....

Reply:
I'm ok with @pmrv 's solution. I agree that the current name is quite clumsy -> it should be something that implies that it prepares the simulation based on the input and returns whether the setup is is correct or not (no idea what it should be called though XD).

At the end of the day, what must be clear is the following routine:

User input -> Compilation -> Execution -> Output collection -> User output

And `check_ready` would correspond to the compilation layer in this flow. In the current pyiron implementation, we have several functions for this, i.e. `check_setup`, `validate_ready_to_run` and `write_input`. But as I already said, I'm also ok with having only one function for this.